pkg/pipeline/pipeline.go

```diff
@@ -575,6 +575,8 @@ type Asset struct {
 
 	upstream   []*Asset
 	downstream []*Asset
+
+	CondtionalColumns []Column `json:"conditions_columns" yaml:"conditions_columns,omitempty" mapstructure:"conditions_columns"`
```

Comment:
we wouldn't want to store them here, intead let's store them inside the `Upstream` object, in the new `columns` field